### PR TITLE
synaptics_dsx: rely on implicit power management performed by touch firmware

### PR DESCRIPTION
**Pull request by vladimiroltean**
* On some panels (mine included), the following code path:
    fb_notifier_callback()
    synaptics_rmi4_resume()
    synaptics_rmi4_sensor_wake()
    synaptics_rmi4_reinit_device()
    synaptics_rmi4_report_touch()

  there is a 2.2 seconds delay between the reinit_device (enables irqs) and
  the report_touch (irq handler) calls, which means that the panel firmware
  actually takes that much time since brought up from sleep and until actually
  emitting irqs to the host. During that time, the display is on, but
  the touch panel is unresponsive.
* Disabling the calls to synaptics_rmi4_sensor_wake() and
  synaptics_rmi4_sensor_sleep() from within fb_notifier_callback() is
  actually done by the stock synaptics_dsx driver shipped by Lenovo as well.
* A difference would be that they do not alter the NO_SLEEP setting, basically
  leaving it set during device sleep (between a synaptics_rmi4_suspend() and a
  synaptics_rmi4_resume()).
* NO_SLEEP setting prevents touch panel from autonomously sleeping when field
  SLEEP_MODE (bits 1:0) of F01_RMI_Ctrl0 is set to 00 (Normal Operation).
  This is acceptable under normal conditions (touch panel is active only
  as long as display is active) but since we have to keep panel in
  Normal Operation mode at all times, not making use of NO_SLEEP is
  highly preferable.
* It is also possible that synaptics_rmi4_sensor_sleep(), which
  basically moves SLEEP_MODE from Normal Operation (00) to Sensor Sleep (01)
  was slightly redundant in the first place, since regulators that power
  the touchscreen panel are disabled in synaptics_rmi4_suspend() as
  well, via the synaptics_dsx_regulator_enable(rmi4_data, false) call.

Change-Id: Ia7960045fd478d9caf3b9652ccc99a28a8818ad1
Signed-off-by: Vladimir Oltean <olteanv@gmail.com>

### CODE REVIEW

**Comment by vladimiroltean** [comment]
Don't merge yet. I just thought I'd push out the patch.
I can't reproduce this anymore, since there seems to be a bug in lineage-15.1 currently. My tablet doesn't go to deep sleep any longer so I can't confirm yet if the patch still has any effect.

**Comment by vladimiroltean** [comment]
At the very least, I forgot to add a `#define IMPLICIT_PM` in the current patch I sent.
